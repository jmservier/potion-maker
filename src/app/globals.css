@tailwind base;
@tailwind components;
@tailwind utilities;

@layer base {
  :root {
    --background: 0 0% 100%;
    --foreground: 0 0% 3.9%;
    --card: 0 0% 100%;
    --card-foreground: 0 0% 3.9%;
    --popover: 0 0% 100%;
    --popover-foreground: 0 0% 3.9%;
    --primary: 0 0% 9%;
    --primary-foreground: 0 0% 98%;
    --secondary: 0 0% 96.1%;
    --secondary-foreground: 0 0% 9%;
    --muted: 0 0% 96.1%;
    --muted-foreground: 0 0% 45.1%;
    --accent: 0 0% 96.1%;
    --accent-foreground: 0 0% 9%;
    --destructive: 0 84.2% 60.2%;
    --destructive-foreground: 0 0% 98%;
    --border: 0 0% 89.8%;
    --input: 0 0% 89.8%;
    --ring: 0 0% 3.9%;
    --radius: 0.75rem;
  }
}

@layer base {
  * {
    @apply border-border;
  }
  body {
    @apply font-sans text-slate-deeper;
    background: linear-gradient(135deg, #f8fafc 0%, #f1f5f9 50%, #e2e8f0 100%);
    min-height: 100vh;
    position: relative;
  }

  body::before {
    content: "";
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background:
      radial-gradient(
        circle at 20% 80%,
        rgba(167, 139, 250, 0.03) 0%,
        transparent 50%
      ),
      radial-gradient(
        circle at 80% 20%,
        rgba(251, 191, 36, 0.03) 0%,
        transparent 50%
      ),
      radial-gradient(
        circle at 40% 40%,
        rgba(94, 234, 212, 0.02) 0%,
        transparent 50%
      );
    pointer-events: none;
    z-index: -1;
  }

  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    @apply font-serif;
  }
}

/* Subtle magical styles */
.glass-card {
  background: rgba(255, 255, 255, 0.85);
  backdrop-filter: blur(12px);
  border: 1px solid rgba(167, 139, 250, 0.1);
  box-shadow: 0 4px 20px rgba(0, 0, 0, 0.05);
}

.ingredient-card {
  background: rgba(255, 255, 255, 0.95);
  border: 1px solid rgba(226, 232, 240, 0.8);
  transition: all 0.3s ease;
  overflow: hidden;
}

.ingredient-card:hover {
  transform: translateY(-2px);
  box-shadow: 0 8px 25px rgba(167, 139, 250, 0.1);
  border-color: rgba(167, 139, 250, 0.3);
  background: rgba(255, 255, 255, 1);
}

.ingredient-card.selected {
  background: linear-gradient(
    135deg,
    rgba(139, 92, 246, 0.1) 0%,
    rgba(167, 139, 250, 0.05) 100%
  );
  border: 2px solid rgba(139, 92, 246, 0.4);
  box-shadow: 0 4px 20px rgba(139, 92, 246, 0.2);
  transform: translateY(-2px);
}

.ingredient-card.selected:hover {
  border-color: rgba(139, 92, 246, 0.6);
  box-shadow: 0 8px 30px rgba(139, 92, 246, 0.3);
}

.ingredient-image {
  width: 100%;
  height: 120px;
  object-fit: cover;
  border-radius: 0.5rem;
  background: linear-gradient(
    135deg,
    rgba(248, 247, 255, 0.5) 0%,
    rgba(243, 229, 171, 0.3) 100%
  );
}

.brewing-area {
  background: linear-gradient(
    135deg,
    rgba(248, 247, 255, 0.3) 0%,
    rgba(243, 229, 171, 0.2) 100%
  );
  border: 2px dashed rgba(167, 139, 250, 0.3);
  transition: all 0.3s ease;
}

.brewing-area.active {
  border-color: rgba(167, 139, 250, 0.5);
  background: linear-gradient(
    135deg,
    rgba(248, 247, 255, 0.5) 0%,
    rgba(243, 229, 171, 0.3) 100%
  );
  box-shadow: inset 0 0 20px rgba(167, 139, 250, 0.1);
}

.potion-bottle {
  background: rgba(255, 255, 255, 0.9);
  border: 1px solid rgba(226, 232, 240, 0.6);
}

.nav-button {
  background: rgba(255, 255, 255, 0.7);
  border: 1px solid rgba(226, 232, 240, 0.8);
  color: #475569;
  transition: all 0.2s ease;
  backdrop-filter: blur(8px);
  font-weight: 500;
}

.nav-button:hover {
  background: linear-gradient(
    135deg,
    rgba(139, 92, 246, 0.15) 0%,
    rgba(124, 58, 237, 0.1) 100%
  );
  border-color: rgba(139, 92, 246, 0.6);
  color: #7c3aed;
  transform: translateY(-2px);
  box-shadow: 0 8px 25px rgba(139, 92, 246, 0.2);
}

.nav-button.active {
  background: linear-gradient(135deg, #a78bfa 0%, #8b5cf6 100%);
  color: white;
  border-color: #a78bfa;
  box-shadow: 0 2px 10px rgba(167, 139, 250, 0.3);
}

.stats-card {
  background: rgba(255, 255, 255, 0.8);
  border: 1px solid rgba(226, 232, 240, 0.6);
  transition: all 0.2s ease;
  backdrop-filter: blur(8px);
}

.stats-card:hover {
  border-color: rgba(167, 139, 250, 0.3);
  box-shadow: 0 4px 15px rgba(0, 0, 0, 0.08);
  background: rgba(255, 255, 255, 0.95);
}

.fade-in {
  animation: fade-in 0.4s ease-out;
}

.scale-in {
  animation: scale-in 0.3s ease-out;
}

.warm-glow {
  box-shadow: 0 4px 20px rgba(0, 0, 0, 0.05);
}

.warm-glow:hover {
  box-shadow: 0 6px 25px rgba(167, 139, 250, 0.1);
}

.gradient-text {
  background: linear-gradient(135deg, #475569 0%, #a78bfa 50%, #fbbf24 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  background-clip: text;
}

.selected-ingredient {
  background: rgba(248, 247, 255, 0.9);
  border: 1px solid rgba(167, 139, 250, 0.4);
  box-shadow: 0 2px 8px rgba(167, 139, 250, 0.15);
}

.btn-primary {
  background: linear-gradient(135deg, #a78bfa 0%, #8b5cf6 100%);
  border: none;
  color: white;
  font-weight: 500;
  transition: all 0.2s ease;
}

.btn-primary:hover {
  background: linear-gradient(135deg, #8b5cf6 0%, #7c3aed 100%);
  transform: translateY(-1px);
  box-shadow: 0 4px 15px rgba(167, 139, 250, 0.3);
}

.btn-secondary {
  background: rgba(255, 255, 255, 0.8);
  border: 1px solid rgba(167, 139, 250, 0.3);
  color: #475569;
  backdrop-filter: blur(8px);
  font-weight: 500;
}

.btn-secondary:hover {
  background: rgba(248, 247, 255, 0.9);
  border-color: rgba(167, 139, 250, 0.5);
}

.gentle-bounce {
  animation: gentle-bounce 2s ease-in-out infinite;
}

@keyframes gentle-bounce {
  0%,
  100% {
    transform: translateY(0px);
  }
  50% {
    transform: translateY(-3px);
  }
}
