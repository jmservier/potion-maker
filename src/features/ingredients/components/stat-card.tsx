interface StatCardProps {
  value: number | string;
  label: string;
}

export function StatCard({ value, label }: StatCardProps) {
  return (
    <div className="stats-card rounded-xl p-6 text-center">
      <div className="mb-1 text-3xl font-bold">{value}</div>
      <div className="text-sm">{label}</div>
    </div>
  );
}
