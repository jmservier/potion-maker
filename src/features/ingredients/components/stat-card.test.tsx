import { describe, expect, it } from "@jest/globals";
import { render } from "@testing-library/react";
import { StatCard } from "./stat-card";

describe("StatCard", () => {
  it("should render correctly with sample props", () => {
    const { container } = render(
      <StatCard value={42} label="Total Ingredients" />,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render correctly with string value", () => {
    const { container } = render(
      <StatCard value="100%" label="Success Rate" />,
    );

    expect(container).toMatchSnapshot();
  });
});
