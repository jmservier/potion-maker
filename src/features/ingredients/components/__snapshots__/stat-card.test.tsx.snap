// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`StatCard should render correctly with sample props 1`] = `
<div>
  <div
    class="stats-card rounded-xl p-6 text-center"
  >
    <div
      class="mb-1 text-3xl font-bold"
    >
      42
    </div>
    <div
      class="text-sm"
    >
      Total Ingredients
    </div>
  </div>
</div>
`;

exports[`StatCard should render correctly with string value 1`] = `
<div>
  <div
    class="stats-card rounded-xl p-6 text-center"
  >
    <div
      class="mb-1 text-3xl font-bold"
    >
      100%
    </div>
    <div
      class="text-sm"
    >
      Success Rate
    </div>
  </div>
</div>
`;
